File: 프로그래머스/3/12938. 최고의 집합/최고의 집합.java
import java.util.*;

class Solution {
    public int[] solution(int n, int s) {
        int[] answer = new int[n];
        Arrays.fill(answer, s / n);
        
        int remain = s - s / n * n;
        int idx = n - 1;
        while (remain-- > 0)
            answer[idx--]++;
        
        if (answer[0] == 0)
            return new int[] {-1};
        return answer;
    }
}

File: 프로그래머스/3/12938. 최고의 집합/최고의 집합.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int n, int s) {
    int div = s / n;
    vector<int> answer(n, s / n);

    if (s % n == 0)
        return answer;

    else {
        int rem = s - s / n * n;
        int idx = answer.size() - 1;
        while (rem--)
            answer[idx--]++;
    }
    if (answer[0])
        return answer;
    return {-1};
}